src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -0,0 +1,232 @@
+
+            throw $e;
+        }
+    }
+
+    public function ack($id): bool
+    {
+        return $this->driverConnection->delete($this->configuration['table_name'], ['id' => $id]) > 0;
+    }
+
+    public function reject($id): bool
+    {
+        return $this->driverConnection->delete($this->configuration['table_name'], ['id' => $id]) > 0;
+    }
+
+    public function requeueMessages()
+    {
+        $redeliverLimit = new \DateTime('now');
+        $redeliverLimit->modify(sprintf('-%d second', $this->configuration['redeliver_timeout']));
+        $sql = $this->driverConnection->createQueryBuilder()
+            ->update($this->configuration['table_name'])
+            ->set('delivered_at', ':delivered_at')
+            ->andWhere('delivered_at < :redeliver_limit')
+            ->andWhere('queue = :queue')
+            ->getSQL();
+
+        $this->executeQuery($sql, [
+            ':delivered_at' => null,
+            ':queue' => $this->configuration['queue'],
+            ':redeliver_limit' => $redeliverLimit->format('Y-m-d H:i:s'),
+        ]);
+    }
+
+    private function executeQuery(string $sql, array $parameters = [])
+    {
+        $stmt = null;
+        try {
+            $stmt = $this->driverConnection->prepare($sql);
+            $stmt->execute($parameters);
+        } catch (TableNotFoundException $e) {
+            // create table
+            if (!$this->driverConnection->isTransactionActive() && $this->configuration['auto_setup']) {
+                $this->setup();
+            }
+            // statement not prepared ? SQLite throw on exception on prepare if the table does not exist
+            if (null === $stmt) {
+                $stmt = $this->driverConnection->prepare($sql);
+            }
+            $stmt->execute($parameters);
+        }
+
+        return $stmt;
+    }
+
+    private function setup(): void
+    {
+        $synchronizer = new SingleDatabaseSynchronizer($this->driverConnection);
+        $synchronizer->updateSchema($this->getSchema(), true);
+    }
+
+    private function getSchema(): Schema
+    {
+        $schema = new Schema();
+        $table = $schema->createTable($this->configuration['table_name']);
+        $table->addColumn('id', Type::BIGINT)
+            ->setAutoincrement(true)
+            ->setNotnull(true);
+        $table->addColumn('body', Type::TEXT)
+            ->setNotnull(true);
+        $table->addColumn('headers', Type::JSON)
+            ->setNotnull(true);
+        $table->addColumn('queue', Type::STRING)
+            ->setLength(32)
+            ->setNotnull(true);
+        $table->addColumn('created_at', Type::DATETIME)
+            ->setLength(3)
+            ->setNotnull(true);
+        $table->addColumn('available_at', Type::DATETIME)
+            ->setLength(3)
+            ->setNotnull(true);
+        $table->addColumn('delivered_at', Type::DATETIME)
+            ->setLength(3)
+            ->setNotnull(false);
+        $table->setPrimaryKey(['id']);
```

Comment:
Should we add an index on some of these columns? I'm really not sure if we should - the table shouldn't get long in theory, but we are doing queries on `delivered_at`, `available_at` and `queue`